src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IfStatement.cs

```diff
@@ -59,7 +59,7 @@ public override BoundNode VisitIfStatement(BoundIfStatement node)
 
                 builder.Add(new BoundConditionalGoto(rewrittenCondition.Syntax, rewrittenCondition, false, afterif));
                 builder.Add(rewrittenConsequence);
-                builder.Add(new BoundLabelStatement(syntax, afterif));
+                builder.Add(new BoundSequencePoint(syntax: null, statementOpt: new BoundLabelStatement(syntax, afterif)));
```

Comment:
There was indeed a problem with the `else` in a conditional, but I didn't find any problems with while or for loops.